### ISSUE TITLE
False negative on `ForEachOnRange`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6163]
```kotlin
@Nested
inner class `using foreach on range with braces or call notation` {
    val code = """
        @OptIn(ExperimentalStdlibApi::class)
        fun test() {
            (1.until(10)).forEach {
                println(it)
            }
            (10.downTo(1)).forEach {
                println(it)
            }
            (10.downTo(1).step(2)).forEach {
                println(it)
            }
            (1.until(10).reversed()).forEach {
                println(it)
            }
            ((1 until 10).reversed()).forEach {
                println(it)
            }
        }
    """.trimIndent()

    @Test
    fun `should report the forEach usage`() {
        val findings = subject.compileAndLint(code)
        assertThat(findings).hasSize(5)
    }
}
```

## Expected Behavior
Above TC should pass.

## Observed Behavior
Above TC is failing

## Context
It seems that when we deviate from the `infix` syntax the code fails to analyse that

**Comment by atulgpt** [comment]
More false negative
```
(1..10).reversed().forEach { 
    println(it)
}
(1..10).step(2).forEach { 
    println(it)
}
```
